### PR TITLE
feat(derive): Test Utilities

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Introduces test utilities to the `derive` crate `traits` module to allow for clean `Stage` unit testing.

### CODE REVIEW

**Comment by clabby** [comment]
👀

File: crates/derive/src/traits/data_sources.rs [hunk 1]
```diff
@@ -21,16 +21,16 @@ pub trait ChainProvider {
...
     /// Returns the data availability for the block with the given hash, or an error if the block does not exist in the
     /// data source.
-    async fn open_data<T: Into<Bytes>>(
+    async fn open_data(
         &self,
         block_ref: &BlockInfo,
         batcher_address: Address,
-    ) -> Result<Self::DataIter<T>>;
+    ) -> Result<Self::DataIter>;
 }
```
Comment (refcell) [inline crates/derive/src/traits/data_sources.rs:34]: This is a minor change I'd like your review on @clabby

**Review by clabby — APPROVED** [review]
Looks great. Let's get this in so we can keep iterating on tests 🚀

**Review by clabby — APPROVED** [review]

File: crates/derive/src/traits/data_sources.rs (hunk 1 above)
Comment (clabby) [reply crates/derive/src/traits/data_sources.rs:34]: This looks fine.

**Pull request merged**